plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/DeltaLakeSchemaSupport.java

```diff
@@ -259,9 +259,11 @@ private static Map<String, Object> serializeStructType(ColumnMappingMode columnM
         fields.put("type", "struct");
         fields.put("fields", rowType.getFields().stream()
                 .map(field -> {
+                    String fieldName = field.getName()
+                            .orElseThrow(() -> new IllegalArgumentException("Row field must be named when creating table or adding column"));
                     Object fieldType = serializeColumnType(columnMappingMode, maxColumnId, field.getType());
                     Map<String, Object> metadata = generateColumnMetadata(columnMappingMode, maxColumnId);
-                    return serializeStructField(field.getName().orElse(null), fieldType, null, null, metadata);
+                    return serializeStructField(fieldName, fieldType, null, null, metadata);
```

Comment:
Please add tests in TestDeltaLakeConnectorTest & TestDeltaLakeDatabricksCreateTableCompatibility. 
* TestDeltaLakeConnectorTest should verify the failure of Trino statements
* TestDeltaLakeDatabricksCreateTableCompatibility should verify the Spark behavior